translate/storage/php.py

```diff
@@ -216,171 +339,88 @@ def serialize(self, out):
-                    newunit.escape_type = valuequote
-                    lastvalue = ""
-                    invalue = False
-
-                # If there is more text (a comment) after the translation.
-                if not invalue and enddelpos != (len(value) - 1):
-                    commentinlinepos = value.find("//", enddelpos)
-                    if commentinlinepos != -1:
-                        newunit.addnote(value[commentinlinepos+2:].strip(),
-                                        "developer")
-
-                # If the translation is already parsed, save it and initialize
-                # a new translation unit.
-                if not invalue:
-                    self.addunit(newunit)
-                    value = ""
-                    newunit = phpunit()
-
-                # Update end delimiter position to the previous last appearance
-                # of the end delimiter, because it might be several entries in
-                # the same line.
-                enddelpos = value.rfind(enddel, 0, enddelpos)
-            else:
-                # After processing current line, if we are not in an array,
-                # fall back to default dialect (PHP simple variable syntax).
-                if not inarray:
-                    equaldel = "="
-                    enddel = ";"
-
-            # If this is part of a multiline translation, just append it to the
-            # previous translation lines.
-            if invalue:
-                lastvalue = lastvalue + value + "\n"
+        def newunit(name, value, escape_type, comments):
+            newunit = phpunit()
+            newunit.escape_type = escape_type
+            newunit.addlocation(name)
+            newunit.source = value
+            for comment in comments:
+                newunit.addnote(comment, 'developer')
+            self.addunit(newunit)
+
+        def format_key(key):
+            if isinstance(key, (int, float)):
+                return '{0}'.format(key)
+            return '\'{0}\''.format(key)
+
+        def handle_array(prefix, nodes, lexer):
+            for item in nodes:
+                assert isinstance(item, ArrayElement)
+                # Skip empty keys
+                if item.key == '':
+                    continue
+                name = format_key(item.key)
+                if prefix:
+                    name = '{0}->{1}'.format(prefix, name)
+                if isinstance(item.value, Array):
+                    handle_array(name, item.value.nodes, lexer)
+                elif isinstance(item.value, six.string_types):
+                    newunit(
+                        name,
+                        item.value,
+                        lexer.extract_quote(),
+                        lexer.extract_comments(item.lexpositions[1]),
+                    )
+
+        def concatenate(item):
+            if isinstance(item, six.string_types):
+                return item
+            assert isinstance(item, BinaryOp)
```

Comment:
Is this `assert` necessary? If not please remove.

Reply:
It's there merely to catch in case parser produces something wrong. Therefore I've used assert which would discover this while running tests, but it's situation which really should not happen.